rpcs3/Emu/Cell/Modules/cellMic.cpp

```diff
@@ -3,180 +3,381 @@
 #include "Emu/Cell/PPUModule.h"
 
 #include "cellMic.h"
+#include <Emu/IdManager.h>
+#include <Emu/Cell/lv2/sys_event.h>
+#include <thread>
 
 logs::channel cellMic("cellMic");
 
+void mic_thread::on_init(const std::shared_ptr<void>& _this)
+{
+	named_thread::on_init(_this);
+}
+
+void mic_thread::on_task()
+{
+	while (fxm::check<mic_thread>() && !Emu.IsStopped())
+	{
+		if (Emu.IsPaused())
+		{
+			std::this_thread::sleep_for(1ms); // hack from cellAudio
+			continue;
+		}
+		if (!micOpened || !micStarted)
+			continue;
+
+		// If event queue is not set, then we can't send any events
+		if (eventQueueKey == 0)
+			continue;
+
+		std::this_thread::sleep_for(1s);
```

Comment:
out of curisoitiy, why a full second?

Reply:
also thread state check should happen right after the sleep.

Reply:
I decided to go for a full second because I don't want the thread taking up time on the cpu when it's not really doing anything.